src/expr/src/scalar/func.rs

```diff
@@ -4512,14 +4515,18 @@ fn split_part<'a>(datums: &[Datum<'a>]) -> Result<Datum<'a>, EvalError> {
     ))
 }
 
+fn is_like_pattern_match_static<'a>(a: Datum<'a>, needle: &like_pattern::Matcher) -> Datum<'a> {
+    let haystack = a.unwrap_str();
+    Datum::from(needle.is_match(haystack))
+}
+
 fn is_like_pattern_match_dynamic<'a>(
     a: Datum<'a>,
     b: Datum<'a>,
     case_insensitive: bool,
 ) -> Result<Datum<'a>, EvalError> {
     let haystack = a.unwrap_str();
-    let flags = if case_insensitive { "i" } else { "" };
-    let needle = like_pattern::build_regex(b.unwrap_str(), flags)?;
+    let needle = like_pattern::compile(b.unwrap_str(), case_insensitive, '\\')?;
```

Comment:
Hardcoding this escape char here and in `reduce()` is a little sketch and is a recipe for them getting out of sync. (Honestly the whole `reduce()` structure is a recipe for getting out of sync, since it's _so_ far away from the actual function definitions, but cest le ve.) But! It's fine as long as we do a fast follow and plumb the appropriate regex character in from the SQL ESCAPE clause.